queries.sql

```diff
@@ -93,3 +93,74 @@ SELECT full_name, COUNT(full_name) AS amount FROM owners
 JOIN animals ON animals.owner_id = owners.id
 GROUP BY full_name
 ORDER BY amount DESC LIMIT 1;
+
+
+SELECT a.name animal, vt.name vet, vs.visit_date FROM animals a
+JOIN visits vs ON a.id = vs.animal_id
+JOIN vets vt ON vt.id = vs.vet_id
+WHERE vt.name = 'William Tatcher'
+ORDER BY vs.visit_date DESC LIMIT 1;
+
+
+SELECT ve.name, COUNT(DISTINCT v.animal_id) AS distinct_animals
+FROM visits v
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez'
+GROUP BY ve.name;
+
+
+SELECT vt.name, s.name FROM vets	vt
+LEFT JOIN specializations sp ON vt.id = sp.vet_id
+LEFT JOIN species s ON s.id = sp.species_id;
+
+
+SELECT a.name animal, vt.name vet, vs.visit_date FROM animals a
+JOIN visits vs ON a.id = vs.animal_id
+JOIN vets vt ON vt.id = vs.vet_id
+WHERE 
+	vt.name = 'Stephanie Mendez' AND 
+	vs.visit_date >= '2020-4-1' AND 
+	vs.visit_date <= '2020-08-30';
+
+
+SELECT a.name, COUNT(a.name) amount FROM visits vs
+JOIN animals a ON a.id = vs.animal_id
+GROUP BY a.name
+ORDER BY amount DESC LIMIT 1;
+
+
+SELECT a.name animal_name, vt.name vet, vs.visit_date FROM animals a
+JOIN visits vs ON vs.animal_id = a.id
+JOIN vets vt ON vs.vet_id = vt.id
+WHERE vt.name = 'Maisy Smith'
+ORDER BY vs.visit_date LIMIT 1;
+
+
+SELECT 
+	a.name animal_name,
+	a.date_of_birth animal_birth_date,
+	a.escape_attempts ,
+	a.weight_kg animal_weight,
+	vt.name vet,
+	vt.age vet_age,
+	vt.date_of_graduation,
+	vs.visit_date
+FROM animals a
+JOIN visits vs ON a.id = vs.animal_id
+JOIN vets vt ON vs.vet_id = vt.id
+ORDER BY vs.visit_date DESC
+
```

Comment:
- [ ] It would be nice if you return only the most recent visit means the one visit which is the most recent 👍 to follow the project requirements 😊 because right now you are returning all the visits 😨 but we need only one and the most recent 🤔

Reply:
@HAMMAS-SALEEM Thank you,
![1-1](https://github.com/Alibaba2023/vet-clinic/assets/126192888/c26760b5-6edf-482b-916e-c3ca978f2ee5)

Reply:
Kindly return only the latest (by date) visit not all dear brother